src/audio/generic_processor/generic_processor.c

```diff
@@ -0,0 +1,94 @@
+// SPDX-License-Identifier: BSD-3-Clause
+//
+// Copyright(c) 2020 Intel Corporation. All rights reserved.
+//
+// Author: Marcin Rajwa <marcin.rajwa@linux.intel.com>
+
+/*
+ * A generic processor component.
```

Comment:
Hmm why do you think it is misleding? I don't like the first two proposal from you because they are not appropriate noun. On the other hand generic_codec or even codec_processor would be fine for me.

Reply:
a processor hints at a hardware implementation, at least in English and French.

this really looks like a wrapper for encoder/decoder libraries. Is there any plan to use this for PCM processing?

Reply:
Yes PCM too.

Reply:
ok, so it cannot be using 'codec' 
But it's not generic either, this is only for Cadence libraries, right? so what about cadence_adapter?

Reply:
Not only for Cadence. Also there is a lot of confusion, for example the word `codec` suggest audio stream encoding/decoding functionality however the Cadence codecs can also work with PCM streams. Therefore I feel like `codecq is fine here. More so, they say their API is "common for HiFi audio codec libraries" so we have another keyword "hifi" used here which has many other connotations.

Reply:
@mrajwa please elaborate: who would be using this API? I understand it's use is mainly/only for interfacing with closed-source libraries from Cadence. What others are you referring to?